test/e2e/parallel/nstemplatetier_test.go

```diff
@@ -303,3 +307,132 @@ func TestKsctlGeneratedTiers(t *testing.T) {
 		})
 	}
 }
+
+func TestFeatureToggles(t *testing.T) {
+	t.Parallel()
+	awaitilities := WaitForDeployments(t)
+	hostAwait := awaitilities.Host()
+	memberAwait := awaitilities.Member1()
+
+	base1nsTier, err := hostAwait.WaitForNSTemplateTier(t, "base1ns")
+	require.NoError(t, err)
+
+	t.Run("provision space with enabled feature", func(t *testing.T) {
+		// given
+
+		// Create a new tier which is a copy of base1ns but with an additional ClusterRoleBinding object with "test-feature" annotation.
+		// "feature-test" feature is defined in the ToolchainConfig and has 100 weight
+		tier := tiers.CreateCustomNSTemplateTier(t, hostAwait, "ftier", base1nsTier)
+		createdTier, err := hostAwait.WaitForNSTemplateTier(t, tier.Name)
+		tier = tiers.UpdateCustomNSTemplateTier(t, hostAwait, tier, withClusterRoleBinding(t, createdTier, "test-feature"))
+		require.NoError(t, err)
+
+		// when
+
+		// Now let's create a Space
+		user := NewSignupRequest(awaitilities).
+			Username("featured-user").
+			Email("featured@domain.com").
+			ManuallyApprove().
+			TargetCluster(awaitilities.Member1()).
+			EnsureMUR().
+			SpaceTier("base1ns").
+			TargetCluster(memberAwait).
+			RequireConditions(wait.ConditionSet(wait.Default(), wait.ApprovedByAdmin())...).
+			Execute(t)
+		// and promote that space to the ftier tier
+		tiers.MoveSpaceToTier(t, hostAwait, "featured-user", tier.Name)
```

Comment:
should we create a space with the base1ns tier and check that is doesn't have the object expected for the ftier ?

Reply:
We could but wouldn't it be an overkill? The object is defined in the custom tier only. We don't check if objects in appstudio tiers are not ended up in spaces with base1ns tier, for example.
What we could test here is promoting a space with tier `featuredTier` to `base1ns` and see that the object is deleted. But I wanted to keep the e2e-tests low-profile and test the main scenario: enable a feature, disable a feature. And leave all the corner cases to unit tests.
Do you think we should have a test for the promotion case in e2e-test anyway?

Reply:
> We could but wouldn't it be an overkill? The object is defined in the custom tier only. We don't check if objects in appstudio tiers are not ended up in spaces with base1ns tier, for example.

that's right, we can avoid this indeed 👍 

> Do you think we should have a test for the promotion case in e2e-test anyway?

If we think it's a scenario that would be frequent yes, if it's just a corner case - I mean if we will not perform this promotion from featured tier to non featured tiers often then it's not worth adding e2e tests ( increasing overall execution time and adding more code to maintain ) , unit tests covering this would be enough indeed. I don't think we will move users between these tiers often , maybe it's fine to leave it for later if needed 👍